src/physics/em/BetheHeitlerInteractor.i.hh

```diff
@@ -0,0 +1,240 @@
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file BetheHeitlerInteractor.i.hh
+//---------------------------------------------------------------------------//
+#include <iostream>
+#include "base/ArrayUtils.hh"
+#include "base/Constants.hh"
+#include "random/distributions/BernoulliDistribution.hh"
+#include "random/distributions/GenerateCanonical.hh"
+#include "random/distributions/UniformRealDistribution.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with shared and state data.
+ *
+ * The incident particle must be above the energy threshold: this should be
+ * handled in code *before* the interactor is constructed.
+ */
+BetheHeitlerInteractor::BetheHeitlerInteractor(
+    const BetheHeitlerInteractorPointers& shared,
+    const ParticleTrackView&              particle,
+    const Real3&                          inc_direction,
+    SecondaryAllocatorView&               allocate,
+    const ElementView&                    element)
+    : shared_(shared)
+    , inc_energy_(particle.energy().value())
+    , inc_direction_(inc_direction)
+    , allocate_(allocate)
+    , element_(element)
+{
+    REQUIRE(particle.def_id() == shared_.gamma_id);
+    REQUIRE(inc_energy_ >= this->min_incident_energy()
+            && inc_energy_ <= this->max_incident_energy());
+
+    epsilon0_ = 1.0 / (shared_.inv_electron_mass * inc_energy_.value());
+    CHECK(epsilon0_ < 1.0);
+    static_assert(sizeof(real_type) == sizeof(double),
+                  "Embedded constants are hardcoded to double precision.");
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Pair-production using the Bethe-Heitler model.
+ *
+ * See section 6.5 of the Geant physics reference 10.6.
+ */
+template<class Engine>
+CELER_FUNCTION Interaction BetheHeitlerInteractor::operator()(Engine& rng)
+{
+    // Allocate space for the pair-produced electrons
+    Secondary* electron_pair = this->allocate_(2);
+    if (electron_pair == nullptr)
+    {
+        // Failed to allocate space for a secondary
+        return Interaction::from_failure();
+    }
+
+    // Construct interaction for change to primary (incident) particle
+    Interaction result = Interaction::from_absorption();
+    result.secondaries = {electron_pair, 2};
+
+    // Outgoing secondaries are electron and positron
+    electron_pair[0].def_id = shared_.electron_id;
+    electron_pair[1].def_id = shared_.positron_id;
+
+    // If E_gamma < 2 MeV, rejection not needed -- just sample uniformly
+    real_type epsilon;
+    if (inc_energy_.value() < 2.0)
+    {
+        UniformRealDistribution<real_type> sample_eps(epsilon0_, 0.5);
+        epsilon = epsilon0_ + (0.5 - epsilon0_) * sample_eps(rng);
+    }
+    // Otherwise, run rejection
+    else
+    {
```

Comment:
Comments have clearer meaning and are less visually intrusive when they're in the code block they applied to, rather than preceding an `if` or visually separating an `if` from its attached `else`.

Also in the uniform sample, you're calculating `a + (b - a) * [(b - a) * xi + a]`, which is sampling uniformly on an odd uniform range that is *not* `[eps_0, .5]`. 

```suggestion
    real_type epsilon;
    if (inc_energy_.value() < 2.0)
    {
        // Rejection is not needed at low energies: just sample uniformly
        UniformRealDistribution<real_type> sample_eps(epsilon0_, 0.5);
        epsilon = sample_eps(rng);
    }
    else
    {
        // Sample with rejection
```